framework/include/base/MooseApp.h

```diff
@@ -950,6 +963,16 @@ class MooseApp : public ConsoleStreamInterface,
    */
   void removeRelationshipManager(std::shared_ptr<RelationshipManager> relationship_manager);
 
+#ifdef LIBTORCH_ENABLED
+  /**
+   * Function to determine the device which should be used by libtorch on this
+   * application. We use this function to decide what is available on different
+   * builds.
+   * @param device String to describe if a cpu or a gpu should be used.
+   */
+  torch::DeviceType determineLibtorchDeviceType(const std::string & device);
```

Comment:
```suggestion
  torch::DeviceType determineLibtorchDeviceType(const std::string & device) const;
```

Should also be private because it should really only be called within the constructor

Reply:
It is private